### PR TITLE
Add ErrInvalidJSONPathArrayCell

### PR DESCRIPTION
**Pull request by erjiaqing**
### What problem does this PR solve? 

Add new error type.

### What is changed and how it works?

This error type is required in JSON_ARRAY_INSERT

### Check List 

Tests 

 - No code

Code changes

 - Has exported variable/fields change

New ErrCode and ErrName is added:
JSON_ARRAY_INSERT
code 3165
name: `A path expression is not a path to a cell in an array.`

Ref.:
https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html#error_er_invalid_json_path_array_cell

>  Error number: 3165; Symbol: ER_INVALID_JSON_PATH_ARRAY_CELL; SQLSTATE: 42000
>
> Message: A path expression is not a path to a cell in an array.

```
MySQL [(none)]> select json_array_insert('{"a": ["b"]}', '$.a', "1");
ERROR 3165 (42000): A path expression is not a path to a cell in an array.
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Where is the TiDB PR that use this? @erjiaqing

**Comment by erjiaqing** [comment]
Still writing, for function JSON_ARRAY_INSERT

**Comment by ngaut** [comment]
LGTM

**Comment by kennytm** [comment]
Please resolve merge conflict @erjiaqing

**Comment by tiancaiamao** [comment]
Ping @erjiaqing

File: mysql/errcode.go [hunk 1]
```diff
@@ -896,6 +896,7 @@ const (
 	ErrInvalidJSONPathWildcard                                      = 3149
 	ErrInvalidJSONContainsPathType                                  = 3150
 	ErrJSONUsedAsKey                                                = 3152
+	ErrInvalidJSONPathArrayCell                                     = 3165
```
Comment (shenli) [inline mysql/errcode.go]: Please move this line after line 900 to keep the error code number in asc order.

**Pull request merged**